File: src/main/java/org/example/data/Authentication.java
package org.example.data;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

@DatabaseTable(tableName = "authentications")
public class Authentication {
    @DatabaseField(generatedId = true)
    public int id;
    @DatabaseField(canBeNull = false, foreign = true, foreignAutoRefresh = true)
    public User user;
    @DatabaseField(canBeNull = false)
    public String salt;
    @DatabaseField(canBeNull = false)
    public String hashedToken;
    @DatabaseField(canBeNull = false)
    public long expiryUnix;
}

File: src/main/kotlin/data/Authentication.kt
package org.example.data

import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable

@DatabaseTable(tableName = "authentications")
class Authentication {
    @DatabaseField(generatedId = true)
    var id: Int = 0
    @Transient
    @DatabaseField(canBeNull = false, foreign = true, foreignAutoRefresh = true)
    var user: User? = null
    @DatabaseField(canBeNull = false)
    var tokenSalt: String = ""
    @DatabaseField(canBeNull = false)
    var hashedToken: String = ""
    @DatabaseField(canBeNull = false)
    var expiryUnix: Long = 0
}